app/src/main/java/com/amaze/filemanager/ui/activities/MainActivity.java

```diff
@@ -1785,7 +1785,34 @@ public void initializeFabActionViews() {
     FabWithLabelView newFolderFab =
         initFabTitle(R.id.menu_new_folder, R.string.folder, R.drawable.folder_fab);
 
-    floatingActionButton.setOnActionSelectedListener(new FabActionListener(this));
+    floatingActionButton.setOnActionSelectedListener(
+        actionItem -> {
+          MainFragment mainFragment = getCurrentMainFragment();
+
+          if (mainFragment == null) return false;
+
+          String path = mainFragment.getCurrentPath();
+
+          MainFragmentViewModel mainFragmentViewModel = mainFragment.getMainFragmentViewModel();
+
+          if (mainFragmentViewModel == null) return false;
+
+          OpenMode openMode = mainFragmentViewModel.getOpenMode();
+
+          int id = actionItem.getId();
+
+          if (id == R.id.menu_new_folder)
+            mainActivity.mainActivityHelper.mkdir(openMode, path, mainFragment);
+          else if (id == R.id.menu_new_file)
+            mainActivity.mainActivityHelper.mkfile(openMode, path, mainFragment);
+          else if (id == R.id.menu_new_cloud)
+            new CloudSheetFragment()
+                .show(mainActivity.getSupportFragmentManager(), CloudSheetFragment.TAG_FRAGMENT);
+
```

Comment:
Code-wise looks OK. Although I'd prefer original `switch` instead of if-else branches unless it's proven to have speed advantage.

Reply:
I was getting a warning with `switch` case: "Resource IDs will be non-final in Android Gradle Plugin version 8.0, avoid using them in switch case statements". Converting to `if else` was the suggested solution.